### ISSUE TITLE
Misannotations in Swedish-Talbanken

### ISSUE THREAD

**Issue by AleksandrsBerdicevskis** [UniversalDependencies/docs#733]
There are four cases when "då" 'then' has UPOS = SCONJ and XPOS = AB ('adverb'). In all of them, UPOS should be "ADV".

**Comment by jnivre** [comment]
Thanks, Sasha. Will fix.

**Comment by AleksandrsBerdicevskis** [comment]
"reda" (as in "ta/få/ha/hålla reda på") is sometimes tagged as NOUN, and sometimes as ADV, I cannot see any principle behind it.

(Is it OK I am posting different bugs in the same issue? There is probably more to come.)

**Comment by jnivre** [comment]
It is fine to keep all of this in one issue (especially as long as it is concerned with part-of-speech tags). Are you only looking at Talbanken, or will you also consider LinES and PUD?

**Comment by AleksandrsBerdicevskis** [comment]
Talbanken only. Basically, we at Språkbanken want to be able to convert our POS(+MSD) annotation (SUC-style) to UD. 
I know you already did the conversion once, but looking at your mamba2ud.py I see that: a) it relies on mamba annotation to some extent; b) it is tailored to Talbanken, i.e. uses many ad hoc solutions; c) some solutions are clearly outdated and have been fixed later. So I decided it would be more efficient to create the new converter and started doing that, looking at the UD guidelines and Talbanken for reference. If I notice any more inconsistencies, I can report them here. 
Does that make sense?

**Comment by jnivre** [comment]
It makes perfect sense.

**Comment by AleksandrsBerdicevskis** [comment]
Two things about adverbs:
1) There are 81 cases when an ADV has an incoming `amod` relation, which should not be the case, if I understand the guidelines correctly. In most of these cases, `amod` should be changed to `advmod`, but in some of them, the pos tag should be changed from ADV to ADJ (and the XPOS has to be changed, too, but I don't know if you care about that). Example of the latter: sentence sv-ud-test-1098, token 34 (_strängt_).

2) Adverbs that are homonymous with neuter adjectives (_samtidigt_) are typically (but not always) lemmatized in the same way as adjectives. i.e. _samtidig_. But does that make sense? If they are treated as adverbs (_pace_ SAG), they should be lemmatized with -t (_samtidig_ cannot be used as an adverb). An alternative solution would be to treat them as adjectives in the adverbial function, then the lemmatization is reasonable.

**Comment by jnivre** [comment]
Re 2: The lemmatization has been done automatically (using an early version of the Språkbanken pipeline) and has only been partially validated manually. So my guess is that these are all (?) cases where the disambiguation has gone wrong. Do you at Språkbanken generally assign the lemma "samtidigt" (rather than "samtidig") to the adverb "samtidigt"? If yes, then we should do the same in the treebank. Presumably most of this could be fixed automatically by searching for words tagged ADV that end in "-t" and have a lemma without the "-t". Or could anything go wrong here?

**Comment by AleksandrsBerdicevskis** [comment]
No, we are actually also lemmatizing it as "samtidig"! This is not very consistent, since in SUC, which is supposed to be gold standard, the lemmatization is "samtidigt". I suspect this is an error rather than conscious policy, but I can check.
I think your rule should generally work fine, but a) it should not just add -t, but rather use the word as lemma (because of words that end in -d etc.); b) it will not work with comparatives and superlatives (which are also inconsistent, cf. tidigare-tidigare, vidare-vitt, djupare-djupt), but usually do have -t.

**Comment by LarsAhrenberg** [comment]
I follow this discussion with interest as Swedish_Lines is not consistent either with respect to lemmas for adverbs. My current preference is for t-forms as lemmas as there are some words where the meaning of the adverb is somewhat different from that of the corresponding adjective, for example 'riktig(t), väldig(t)' I have noticed a difference between Sparv and Efselab wrt lemmatization of these words, Efselab uses t-forms, Sparv does not.

A related question concerns the comparatives and superlatives of adverbs. As in '_Hon sprang längre än i går'_. I guess the lemma should then be _långt_.

**Comment by AleksandrsBerdicevskis** [comment]
I think that's because Sparv's choice of lemma is constrained by Saldo, and Saldo does not list t-adverbs as separate entries (following SAG).

In my opinion, if _samtidigt_ is considered an adverb (as in SUC), the lemma should be _samtidigt_. If it is considered a neuter adjective (as in Saldo and SAG), the lemma should be _samtidig_. Both decisions make sense, but POS=adverb, lemma=_samtidig_ seems inconsistent. Comparatives and superlatives should have the same lemma as positives, I think.

Thanks for your replies! It's useful to become aware of these discrepancies across (and within) the Swedish resources.

**Comment by AleksandrsBerdicevskis** [comment]
_som_, _när_ och _då_ with XPOS=HA (relative adverb) sometimes have UPOS=SCONJ (which seems correct to me) and sometimes ADV (which does not). I cannot see any principle in the distribution, they all seem subjunctions to me (or "relative adverbs" in SUC's terminology). It also affects syntactic relations (`mark` vs `advmod`)

**Comment by jnivre** [comment]
I suspect that ADV + advmod is used when it is analyzed as introducing a relative clause, rather than an adverbial clause, for example, in "dagen då hon kom" (as opposed to a case like "det regnade, då hon kom"). It is debated (not only for Swedish) whether these are really relative clauses, and there may still be inconsistencies, but I think this is the intended distinction.

**Comment by jnivre** [comment]
This is analogous to the use of locative adverbs introducing relative clauses, for example, in "huset där hon bodde", where there is no corresponding adverbial clause use.

**Comment by AleksandrsBerdicevskis** [comment]
Ah, thanks, I see now. I think there are some inconsistencies (especially wrt _när_), but it's difficult to tell without a thorough comparison.

**Comment by gregarshinov** [comment]
I have a question considering `sent_id = sv-ud-test-188` in Talbanken. The 26th token (sammansatt) is annotated weirdly. It's lemma is "sätta_samman" instead of "sammansatt". Is there any explanation to this annotation?

Also, I do not understand why in `sent_id = sv-ud-test-262`, token 3 (första	en	ADJ) is lemmatized as an article and tagged as an Adjective, and not as an ordinal numeral. Could somebody, please, clarify this?

**Comment by dan-zeman** [comment]
> I follow this discussion with interest as Swedish_Lines is not consistent either with respect to lemmas for adverbs.

Since this issue is prevailingly about (possible) bugs in Talbanken, I am going to move it from the docs issue tracker to the UD_Swedish-Talbanken issue tracker. If there are any open questions about Swedish-specific annotation guidelines, feel free to open a new issue here (in docs) for each such question.
